hls4ml/backends/fpga/passes/codegen.py

```diff
@@ -49,3 +49,25 @@ def _generate_im2col_2d(self, node):
         )
 
         node.set_attr('line_buffer_codegen', Source(code_str))
+
+
+class GeneratePointwiseConv1D(OptimizerPass):
```

Comment:
hmm, I'm thinking if we should split this into a new file? and establish a naming convention. for example, current codegen.py could be im2col_codegen.py and this one pointwise_codegen.py. this will get cramped in the future (e.g., Chang will also use this for HGQ)

Reply:
split the new code into a new file `vivado/passes/pointwise_codegen.py` and renamed this one to `fpga/passes/im2col_codegen.py`